Source/items.cpp

```diff
@@ -3813,60 +3812,47 @@ void PrintItemMisc(ItemStruct *x)
 
 void PrintItemDetails(ItemStruct *x)
 {
-	ItemStruct *v1;   // ebp
-	char v2;          // cl
-	char v3;          // cl
-	char v4;          // al
-	unsigned char v5; // al
-	char v6;          // al
-
-	v1 = x;
 	if (x->_iClass == 1) {
 		if (x->_iMaxDur == 255)
 			sprintf(tempstr, "damage: %i-%i  Indestructible", x->_iMinDam, x->_iMaxDam);
 		else
 			sprintf(tempstr, "damage: %i-%i  Dur: %i/%i", x->_iMinDam, x->_iMaxDam, x->_iDurability, x->_iMaxDur);
 		AddPanelString(tempstr, 1);
 	}
-	if (v1->_iClass == 2) {
-		if (v1->_iMaxDur == 255)
-			sprintf(tempstr, "armor: %i  Indestructible", v1->_iAC);
+	if (x->_iClass == 2) {

```

Comment:
Does iClass have an enum?

Reply:
I had missed it, but it actually does

Reply:
https://github.com/diasurgical/devilution/commit/e7a3109f17b9a7e5180228c108560010f6cbc516